lib/workers/repository/updates/generate.ts

```diff
@@ -436,9 +436,23 @@ export function generateBranchConfig(
         .reduce((a, b) => a.concat(b), []),
     ),
   ];
+
   if (config.upgrades.some((upgrade) => upgrade.updateType === 'major')) {
     config.updateType = 'major';
   }
+
+  // explicit set `isLockFileMaintenance` for the branch for groups
+  if (config.upgrades.some((upgrade) => upgrade.isLockFileMaintenance)) {
```

Comment:
Here, config = branch config. So you're setting branchConfig.isLockFileMaintenance=true if any update in the branch  has isLockFileMaintenance set. This seems logically correct. But I think `updateArtifacts()` is no called on a full branch, so the refactorings within manager code seem unrelated or mismatched with this change?

Reply:
No, it's passing the full branch config. `patchConfigForArtifactsUpdate` is only patching the `lockFiles` property and `config` is of `BranchConfig` type.

https://github.com/renovatebot/renovate/blob/f182708232b5aa623b77f0f3dc7d5e2e002b6567/lib/workers/repository/update/branch/get-updated.ts#L334-L344
https://github.com/renovatebot/renovate/blob/f182708232b5aa623b77f0f3dc7d5e2e002b6567/lib/workers/repository/update/branch/get-updated.ts#L375-L385
https://github.com/renovatebot/renovate/blob/f182708232b5aa623b77f0f3dc7d5e2e002b6567/lib/workers/repository/update/branch/get-updated.ts#L421-L429